src/runtime_src/core/edge/ps_kernels/built_in/instance_query/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2019-2022 Xilinx, Inc. All rights reserved.
+# Copyright (C) 2022 Advanced Micro Devices, Inc. All rights reserved.
+#
+set(SRC_DIR  "${CMAKE_CURRENT_SOURCE_DIR}")
+set(PS_KERNEL_INSTALL_DIR "${XRT_INSTALL_LIB_DIR}/ps_kernels_lib")
+
+include_directories(
+  ${SRC_DIR}
+  )
+
+file(GLOB SRC_FILES
+  "${SRC_DIR}/*.cpp"
+  )
+
+set(CMAKE_CXX_FLAGS "-DXAIE_DEBUG ${CMAKE_CXX_FLAGS}")
+add_definitions(-DXCLHAL_MAJOR_VER=1 -DXCLHAL_MINOR_VER=0)
+add_library(instance_query SHARED ${SRC_FILES})
```

Comment:
Expand SRC_FILES to the files you want included in the library.
Please check other CMakeLists.txt files, e.g. core/common/CMakeLists.txt (probably not perfect either, but better)